extension-style/src/main/java/com/mapbox/maps/extension/style/sources/generated/GeoJsonSource.kt

```diff
@@ -618,30 +692,61 @@ class GeoJsonSource(builder: Builder) : Source(builder.sourceId) {
      * @param value the geometry
      */
     fun geometry(value: Geometry) = apply {
-      rawGeoJson = value
-      val propertyValue = PropertyValue("data", "null")
-      properties[propertyValue.propertyName] = propertyValue
+      setGeoJson(value)
+    }
+
+    private fun setGeoJson(geoJson: GeoJson) {
+      this.geoJson = geoJson
+      if (directSetterEnabled) {
+        data = null
+      } else {
+        val propertyValue = PropertyValue("data", "null")
+        properties[propertyValue.propertyName] = propertyValue
+      }
     }
 
     /**
      * Initialize GeoJsonSource builder with default data to allow empty data source.
      */
     init {
-      this.data("")
+      if (!directSetterEnabled) {
+        this.data("")
+      }
     }
 
     /**
      * Build the GeoJsonSource.
      *
      * @return the GeoJsonSource
      */
-    fun build() = GeoJsonSource(this, rawGeoJson)
+    fun build(): GeoJsonSource {
+      if (directSetterEnabled) {
+        // we need empty initial data when creating the geojson (as we don't set it in constructor) -
+        // actual data (if specified by the user) will be applied asynchronously later
+        val propertyValue = PropertyValue("data", TypeUtils.wrapToValue(""))
+        properties[propertyValue.propertyName] = propertyValue
+      }
+      return GeoJsonSource(this, geoJson, data)
+    }
   }
 
   /**
    * Static variables and methods.
    */
   companion object {
+    private const val TAG = "GeoJsonSource"
+
+    /**
+     * Check if runtime property geojson_allow_direct_setter is enabled.
+     */
+    @VisibleForTesting(otherwise = VisibleForTesting.PRIVATE)
+    fun directSetterEnabled(): Boolean {
+      val settingValue = SettingsServiceFactory
+        .getInstance(SettingsServiceStorageType.NON_PERSISTENT)
+        .get("geojson_allow_direct_setter")
```

Comment:
Should string `"geojson_allow_direct_setter"` be a constant so users can use it directly instead of hardcoded string?

Reply:
That's not really something we want to expose (that's the reason for marking this PR as skip changelog) so exposing constant imo is not needed here.

Reply:
As per plan using direct setter will be introduced fully in v10.11-beta.1